pkg/testutils/e2e/kubectl/kubectl.go

```diff
@@ -177,3 +185,109 @@ func (c *Client) Apply(ctx context.Context, yaml string) error {
 
 	return nil
 }
+
+type portForwardConfig struct {
+	localport uint
+	stderr    io.Writer
+	stdout    io.Writer
+}
+
+// PortForwardOption defines a configuration option for port forwarding
+type PortForwardOption func(portForwardConfig) portForwardConfig
+
+// WithLocalPort sets the local port to listen for request. Defaults to 0 (random local port)
+func WithLocalPort(port uint) PortForwardOption {
+	return func(p portForwardConfig) portForwardConfig {
+		p.localport = port
+		return p
+	}
+}
+
+// WithOutputStreams sets the output streams for the port forwarder. Default to nil
+func WithOutputStreams(stdout io.Writer, stderr io.Writer) PortForwardOption {
+	return func(p portForwardConfig) portForwardConfig {
+		p.stdout = stdout
+		p.stderr = stderr
+		return p
+	}
+}
+
+func newPortForwardConfig(opts ...PortForwardOption) portForwardConfig {
+	config := portForwardConfig{
+		localport: 0,
+		stdout:    nil,
+		stderr:    nil,
+	}
+
+	for _, option := range opts {
+		config = option(config)
+	}
+
+	return config
+}
+
+// ForwardPodPort opens a local port for forwards requests to a pod's port.
+// Returns the local port used for listening
+func (c *Client) ForwardPodPort(
+	ctx context.Context,
+	pod string,
+	namespace string,
+	port uint,
+	opts ...PortForwardOption,
+) (uint, error) {
+	config := newPortForwardConfig(opts...)
+
+	path := fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", namespace, pod)
+	host, err := url.Parse(c.config.Host)
+	if err != nil {
+		return 0, fmt.Errorf("invalid host URL in k8s client config: %w", err)
+	}
+	url := &url.URL{
+		Scheme: "https",
+		Path:   path,
+		Host:   host.Host,
+	}
+
+	transport, upgrader, err := spdy.RoundTripperFor(c.config)
+	if err != nil {
+		return 0, err
+	}
+	dialer := spdy.NewDialer(
+		upgrader,
+		&http.Client{Transport: transport},
+		http.MethodPost,
+		url,
+	)
+
+	ports := []string{fmt.Sprintf("%d:%d", config.localport, port)}
+	ready := make(chan struct{})
+	fw, err := portforward.New(
+		dialer,
+		ports,
+		ctx.Done(),
+		ready,
+		config.stdout,
+		config.stderr,
+	)
+	if err != nil {
+		return 0, err
+	}
+
+	errors := make(chan error)
+	go func() {
+		errors <- fw.ForwardPorts()
+	}()
+
+	// Wait for the port forwarder to be ready and return stop channel
```

Comment:
I don't understand this comment, which stop channel are we returning?

Reply:
Good catch. The comment is outdated.